### PR TITLE
BFD-897: Fix BFD Pipeline's init errors and job failure handling

### PR DESCRIPTION
**Pull request by karlmdavis**
### Change Details

Fixes the exit-on-error issue that caused us to revert the changes initially. Also ensures that job failures are logged appropriately.

### Acceptance Validation

1. Does this fix the intended issues?
1. Does this code look okay?

### Feedback Requested

(see above)

### External References

- [BFD-897](https://jira.cms.gov/browse/BFD-897)

### Security Implications

This PR has no known security implications.

- [ ] new software dependencies

- [ ] altered security controls

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

### CODE REVIEW

**Comment by karlmdavis** [comment]
This is a replacement for #640, which had gone sideways, somehow.

**Comment by karlmdavis** [comment]
And now this one is too, and no rejiggering of the base branches seems to allow me to fix it. I think we're going to have to merge the unrevert first and then this right after I can rebase it against master.

**Review by brianburton — APPROVED** [review]
Look good!

File: apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java [hunk 1]
```diff
@@ -135,7 +135,7 @@
    * Constructs a new {@link AppConfiguration} instance.
    *
    * @param metricOptions the value to use for {@link #getMetricOptions()}
-   * @param databaseOptions the value to use for {@link #getDatabaseOptions()
+   * @param databaseOptions the value to use for {@link #getDatabaseOptions()}
```
Comment (jzulim) [inline apps/bfd-pipeline/bfd-pipeline-app/src/main/java/gov/cms/bfd/pipeline/app/AppConfiguration.java:138]: Good eyes.

**Review by jzulim — APPROVED** [review]
shiiiiiiip it. 🚢 - Thank you for tackling this @karlmdavis and for the walkthrough.

**Pull request merged**